docs/src/components/TableOfContents.tsx

```diff
@@ -1,11 +1,56 @@
 import React from "react";
 import Scrollspy from "react-scrollspy";
 import styled, { css } from "styled-components";
+import { v4 as uuidv4 } from "uuid";
 
-export interface TocItemObject {
+import { slugify } from "../utils/common";
+
+const DEPTH = 3;
+
+interface TableOfContentsItem {
+  id: string;
   title: string;
-  url: string;
-  items?: TocItemObject[];
+  slug: string;
+  level: number;
+}
+
+type SetTableOfContents = React.Dispatch<React.SetStateAction<TableOfContentsItem[]>>;
+
+const TableOfContentsContext = React.createContext<[TableOfContentsItem[], SetTableOfContents]>([
+  [],
+  () => {},
+]);
+TableOfContentsContext.displayName = "TableOfContentsContext";
+
+export function TableOfContentsProvider({ children }: { children: React.ReactNode }) {
+  const [tableOfContents, setTableOfContents] = React.useState<TableOfContentsItem[]>([]);
+  return (
+    <TableOfContentsContext.Provider value={[tableOfContents, setTableOfContents]}>
+      {children}
+    </TableOfContentsContext.Provider>
+  );
+}
+
+export function useTableOfContents() {
+  const [tableOfContents] = React.useContext(TableOfContentsContext);
+  return tableOfContents;
+}
+
+export function useTableOfContentsRegister({ title, level }: { title: string; level: number }) {
+  const [, setTableOfContents] = React.useContext(TableOfContentsContext);
+  const id = React.useMemo(() => uuidv4(), []);
```

Comment:
i think you can use our `randomID` function, if you will pass there title of current item? 🤔